geometry/sign_body.hpp

```diff
@@ -36,20 +69,22 @@ inline void Sign::WriteToMessage(
 }
 
 inline Sign Sign::ReadFromMessage(serialization::Sign const& message) {
-  return Sign(message.negative() ? -1 : 1);
+  return Sign(/*negative=*/message.negative());
 }
 
-inline Sign operator*(Sign const& left, Sign const& right) {
-  return Sign(left.negative_ == right.negative_ ? 1 : -1);
+constexpr Sign::Sign(bool negative) : negative_(negative) {}
+
+constexpr Sign operator*(Sign const& left, Sign const& right) {
+  return Sign(/*negative=*/left.negative_ != right.negative_);
```

Comment:
Use `^` instead of `!=`.

Reply:
`^` is an arithmetic operator, not a logical one, so this requires a cast; from the point of view of readability, do we want that?

In terms of generated code, it does appear to make a difference with MSVC, but only if `^` is applied to signed operands: https://gcc.godbolt.org/z/dGvY55; it makes no difference with clang.